### PR TITLE
Add highlight/query similar features example

### PR DESCRIPTION
**Pull request by tristen**
highlight-similar-features

cc @mourner @lucaswoj @jfirebaugh for review

### CODE REVIEW

File: docs/_posts/examples/3400-01-24-query-similar-features.html [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        "filter": ["in", "COUNTY", ""]
+    }, 'place-city-sm'); // Place polygon under these labels.
+
+    map.on('mousemove', function(e) {
+        var features = map.queryRenderedFeatures(e.point, {
+            layers: ['counties']
+        });
+
+        // Change the cursor style as a UI indicator.
+        map.getCanvas().style.cursor = (features.length) ? 'pointer' : '';
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-24-query-similar-features.html]: No parens around `features.length`.

File: docs/_posts/examples/3400-01-24-query-similar-features.html [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+        // Remove things if no feature was found.
+        if (!features.length) {
+            popup.remove();
+            map.setFilter('counties-highlighted', ['in', 'COUNTY', '']);
+            renderOverlay([]);
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-24-query-similar-features.html]: Inline the "no features" case of `renderOverlay` here.

File: docs/_posts/examples/3400-01-24-query-similar-features.html [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+        // Query the counties layer visible in the map. Use the filter
+        // param to only collect results that share the same county name.
+        var relatedFeatures = map.querySourceFeatures('counties', {
+            sourceLayer: 'original',
+            filter: ['in', 'COUNTY', feature.properties.COUNTY]
+        });
+
+        // Render found features in an overlay.
+        renderOverlay(relatedFeatures);
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-24-query-similar-features.html]: And the "yes features" case here.

Comment (tristen) [inline docs/_posts/examples/3400-01-24-query-similar-features.html]: Not sure I follow this comment. There's an inline comment above this function that says

> Render found features in an overlay.

Comment (tristen) [inline docs/_posts/examples/3400-01-24-query-similar-features.html]: I'm following now :) inline the code I created a named function for. It only happens in one place.

**Comment by tristen** [comment]
:ship:'d https://github.com/mapbox/mapbox-gl-js/commit/5342aa93d4ea111cc910cf68769e3716a6f0aba1

**Comment by nitins1** [comment]
How do I find the data stored in the source file (mapbox://mapbox.82pkq93d) so I can manipulate the contents differently. For example, select counties in certain states as well, but using "STATES" as the key in the filter argument of the setFilter method doesn't work. Any ideas?

**Comment by tristen** [comment]
👋 @nitins1 sounds like you probably want [`querySourceFeatures`](https://www.mapbox.com/mapbox-gl-js/api/#Map#querySourceFeatures)?

**Comment by lbud** [comment]
@nitins1 — @tristen is right in that you can use `querySourceFeatures` to dig into the data added to a GL map.

Alternatively, you can look for that dataset's [tileJSON](https://www.mapbox.com/api-documentation/#retrieve-tilejson-metadata) — in this case, at `https://api.mapbox.com/v4/mapbox.82pkq93d.json?secure&access_token=<your access token>`
which contains this field:
```
  "vector_layers": [
    {
      "description": "",
      "fields": {
        "COUNTY": "String",
        "FIPS": "Number",
        "median-income": "Number",
        "population": "Number"
      },
      "id": "original",
      "maxzoom": 22,
      "minzoom": 0,
      "source": "mapbox.82pkq93d",
      "source_name": "counties"
    }
  ],
```

**Comment by nitins1** [comment]
@tristen @lbud Thank you both for your responses! So I understand that I can use `querySourceFeatures` to return an array of objects that match a given filter. But after looking into the dataset's tileJSON, it doesn't seem like this data source has any information about states or counties relating to a state. Is there a data source/another walkthrough example with state data?

**Pull request closed without merging**